stats/opentelemetry/e2e_test.go

```diff
@@ -46,35 +46,10 @@ func Test(t *testing.T) {
 	grpctest.RunSubTests(t, s{})
 }
 
-// waitForServerCompletedRPCs waits until the unary and streaming stats.End
-// calls are finished processing. It does this by waiting for the expected
-// metric triggered by stats.End to appear through the passed in metrics reader.
-func waitForServerCompletedRPCs(ctx context.Context, reader metric.Reader, wantMetric metricdata.Metrics, t *testing.T) (map[string]metricdata.Metrics, error) {
-	for ; ctx.Err() == nil; <-time.After(time.Millisecond) {
-		rm := &metricdata.ResourceMetrics{}
-		reader.Collect(ctx, rm)
-		gotMetrics := map[string]metricdata.Metrics{}
-		for _, sm := range rm.ScopeMetrics {
-			for _, m := range sm.Metrics {
-				gotMetrics[m.Name] = m
-			}
-		}
-		val, ok := gotMetrics[wantMetric.Name]
-		if !ok {
-			continue
-		}
-		if !metricdatatest.AssertEqual(t, wantMetric, val, metricdatatest.IgnoreTimestamp(), metricdatatest.IgnoreExemplars()) {
-			continue
-		}
-		return gotMetrics, nil
-	}
-	return nil, fmt.Errorf("error waiting for metric %v: %v", wantMetric, ctx.Err())
-}
-
 // setup creates a stub server with OpenTelemetry component configured on client
 // and server side. It returns a reader for metrics emitted from OpenTelemetry
 // component and the server.
-func setup(t *testing.T, tafOn bool, maf func(string) bool) (*metric.ManualReader, *stubserver.StubServer) {
+func setup(t *testing.T, maf func(string) bool) (*metric.ManualReader, *stubserver.StubServer) {
```

Comment:
There is a lot of setup in this function, but it is not very readable.

- what is `maf`? For someone new to the code base, it is not obvious that it the method attribute filter func
- Is it possible to have the stub server initialized within the test, so that the RPC handler functionality is close to the test logic. See: go/go-style/best-practices#t-common-setup-scope
- initialization of the `reader` and the `provider` and one liners, but are done here. And this means that when one reads the test logic, it is not clear what the `reader` returned from this function is, and they will have to navigate to here.

Reply:
I'm not going to ask you to refactor this as part of this PR, but something to keep in mind: https://engdoc.corp.google.com/eng/doc/tott/episodes/643.md?cl=head